support/notifications-client/client.go

```diff
@@ -68,38 +82,42 @@ type Notification struct {
 	Modified    int          `json:"modified,omitempty"` // The last modification timestamp
 }
 
-// Send a POST call to the notifications service
-func (nc NotificationsClient) RecieveNotification(n Notification) error {
+// Send a notification to the notifications service
+func (nc NotificationsClient) SendNotification(n Notification) error {
 	client := &http.Client{}
 
 	// Get the JSON request body
 	requestBody, err := json.Marshal(n)
 	if err != nil {
-		fmt.Println(err.Error())
 		return err
 	}
 
 	// Create the request
-	req, err := http.NewRequest(http.MethodPost, nc.RemoteUrl, bytes.NewBuffer(requestBody))
-	req.Header.Add("Content-Type", "application/json")
+	remoteNotificationServiceUrl := fmt.Sprintf(UrlPattern, nc.NotificationServiceHost, nc.NotificationServicePort, NotificationApiPath)
+
+	return doPost(remoteNotificationServiceUrl, bytes.NewBuffer(requestBody), client)
+}
+
+// Function to do post request
+func doPost(url string, binaryReqBody io.Reader, client *http.Client) error {
+	req, err := http.NewRequest(http.MethodPost, url, binaryReqBody)
+	req.Header.Add(ContentType, ContentTypeJsonVal)
+
 	if err != nil {
-		fmt.Println(err.Error())
 		return err
 	}
 
-	// Asynchronous call
-	go makeRequest(client, req)
-
-	return nil
+	return sendRequest(client, req)
 }
 
 // Function to actually make the HTTP request
-func makeRequest(client *http.Client, req *http.Request) {
+func sendRequest(client *http.Client, req *http.Request) error {
 	resp, err := client.Do(req)
 	if err == nil {
```

Comment:
Strange construct here - rather check `if err != nil`

Reply:
will change this mode by follow the offical : https://golang.org/pkg/net/http/

Reply:
cc @drasko changed this.